extensions/wikia/MercuryApi/MercuryApiHooks.class.php

```diff
@@ -120,44 +103,24 @@ static public function onInstantGlobalsGetVariables( array &$vars ) {
 	}
 
 	static public function onCuratedContentSave( $sections ) {
-		global $wgServer;
-
 		// Purge main page cache, so Mercury gets fresh data.
 		Title::newMainPage()->purgeSquid();
 
 		$urls = [ ];
 		WikiaDataAccess::cachePurge( MercuryApiMainPageHandler::curatedContentDataMemcKey() );
 
 		foreach ( $sections as $section ) {
-			if ( !empty( $section['featured'] ) ) {
+			$sectionLabel = $section['label'];
+
+			if ( empty( $sectionLabel ) || !empty( $section['featured'] ) ) {
 				continue;
 			}
 
-			$sectionTitle = $section['title'];
-
-			WikiaDataAccess::cachePurge( MercuryApiMainPageHandler::curatedContentDataMemcKey( $sectionTitle ) );
-
-			// We have to double encode because Ember's RouteRecognizer does decodeURI while processing path.
-			$doubleEncodedTitle = self::encodeURI( self::encodeURIQueryParam( $sectionTitle ) );
-
-			// Mercury opened directly with URL
-			$urls[] =
-				$wgServer .
-				self::SERVICE_API_ROOT .
-				self::SERVICE_API_CURATED_CONTENT .
-				$doubleEncodedTitle;
-
-			// API request from Ember to Hapi
-			$urls[] =
-				$wgServer .
-				self::SERVICE_API_ROOT .
-				self::SERVICE_API_BASE .
-				self::SERVICE_API_CURATED_CONTENT .
-				$doubleEncodedTitle;
+			WikiaDataAccess::cachePurge( MercuryApiMainPageHandler::curatedContentDataMemcKey( $sectionLabel ) );
 
-			// Request from Hapi to MediaWiki
-			$encodedTitle = self::encodeURIQueryParam( $sectionTitle );
-			$urls[] = MercuryApiController::getUrl( 'getCuratedContentSection', [ 'section' => $encodedTitle ] );
+			// Request from browser to MediaWiki
+			$encodedTitle = self::encodeURIQueryParam( $sectionLabel );
+			$urls[] = MercuryApiController::getUrl( 'getCuratedContentSection' ) . '&section=' . $encodedTitle;
```

Comment:
`getUrl` method does the additional encoding so I used the same solution as is in https://github.com/Wikia/app/blob/dev/extensions/wikia/CuratedContent/CuratedContentHooks.class.php#L30